services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/model/BaseDto.java

```diff
@@ -43,22 +45,56 @@ public BaseDto() {
     }
 
     /**
-     * Gets the version of the document.
+     * Gets the version number of the document.
      * 
+     * @return The version number of the document or {@code null} if not set.
+     */
+    public final long getVersionNo() {
+        return versionNo;
+    }
+
+    /**
+     * Gets the version of the document.
+     *
      * @return The version of the document or {@code null} if not set.
      */
+    @JsonIgnore
     public final String getVersion() {
-        return version;
+        return String.format("%d", versionNo);
```

Comment:
you really want to compute this every time this method is invoked?

Reply:
As I thought through it, FMPOV normally the read operations in device registry are performed more frequently than the write operations. Every time type casting or formatting during read operation is not so efficient. So, I would replace the data type of the `version` to `String`. The typecasting could be performed during add/update operations. WDYT?

Reply:
That is probably true.

Reply:
In many device registration operations, `Future<OperationResult>` is being returned. When an operation is successful `OperationResult.ok(...)` is being used for that and it expects the version to be String. 

```
OperationResult<T> ok(final int status, final T payload,
            final Optional<CacheDirective> directive, final Optional<String> resourceVersion)
```

Reply:
I see. So let's go with a String valued version field then and do the serialization as part of the setter, ok?